lib/BuildSystem/SwiftTools.cpp

```diff
@@ -624,8 +630,322 @@ class SwiftCompilerTool : public Tool {
     return nullptr;
   }
 };
+
+class SwiftCompilerOutputMessage {
```

Comment:
After looking at this some more, I think it should probably move to a parser + delegate model. The parser would take in the incoming stream of bytes, and it would take a delegate with a bunch of methods like:
```
virtual void actOnBegin( ... )
virtual void actOnFinished(...)
```

The reason this makes more sense if that the clients of this API are probably going to have very specialized logic for each case. It doesn't make sense to pack up all the data into a uniform structure just to turn around and have the clients unpack it.

This would also have another major advantage of being unit testable, which I would recommend actually doing. If you want an example, look at 600d1262d019003c4dbef8214224556145fdb796.